.github/workflows/coverage.yml

```diff
@@ -0,0 +1,45 @@
+on:
+  push:
+    branches: [master]
+  pull_request:
+name: coverage
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container:
+      image: xd009642/tarpaulin
+      options: --security-opt seccomp=unconfined
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          chmod 600 .test-key
+          ssh -i .test-key -v -p 2222 -l test-user openssh -o StrictHostKeyChecking=accept-new whoami
+        name: Test ssh connectivity
+      - run: |
+          eval $(ssh-agent)
+          echo "SSH_AUTH_SOCK=$SSH_AUTH_SOCK" >> $GITHUB_ENV
+          echo "SSH_AGENT_PID=$SSH_AGENT_PID" >> $GITHUB_ENV
+          cat .test-key | ssh-add -
+        name: Set up ssh-agent
+      - name: Generate code coverage
+        run: |
+          cargo tarpaulin --verbose --timeout 300 --out Xml --forward --all-features
+        env:
+          # makes all the ignored tests not ignored
+          RUSTFLAGS: --cfg=ci
+          # we cannot use 127.0.0.1 (the default here)
+          # since we are running from a different container
+          TEST_HOST: ssh://test-user@openssh:2222
+      - name: Upload to codecov.io
+        uses: codecov/codecov-action@v2
+        with:
+          fail_ci_if_error: true
+    services:
+      openssh:
+        image: linuxserver/openssh-server:amd64-latest
+        ports:
+          - 2222:2222
+        env:
+          USER_NAME: test-user
```

Comment:
`linuxserver/openssh-server` does not enable ssh port forwarding by default, it requires an additional environment variable to be set:

```
DOCKER_MODS='linuxserver/mods:openssh-server-ssh-tunnel'
```